lib/resty/websocket/client.lua

```diff
@@ -161,6 +168,15 @@ function _M.connect(self, uri, opts)
         return 1
     end
 
+    local custom_headers
+    if headers then
+      local tmp = {}
+      for k, v in pairs(headers) do
+        tmp[#tmp + 1] = "\r\n" .. k .. ": " .. tostring(v)
```

Comment:
Again, `#` on Lua tables is an `O(n)` operation. It's especially bad when being used in a loop, which leads to `O(n^2)` for the whole loop.

Reply:
Fixed.